bigtable-client-core-parent/bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableOptionsFactory.java

```diff
@@ -550,11 +594,452 @@ private static RetryOptions createRetryOptions(Configuration configuration) {
     retryOptionsBuilder.setStreamingBufferSize(streamingBufferSize);
 
     int maxScanTimeoutRetries =
-        configuration.getInt(
-            MAX_SCAN_TIMEOUT_RETRIES, RetryOptions.DEFAULT_MAX_SCAN_TIMEOUT_RETRIES);
+        configuration.getInt(MAX_SCAN_TIMEOUT_RETRIES, DEFAULT_MAX_SCAN_TIMEOUT_RETRIES);
     LOG.debug("gRPC max scan timeout retries (count): %d", maxScanTimeoutRetries);
     retryOptionsBuilder.setMaxScanTimeoutRetries(maxScanTimeoutRetries);
 
     return retryOptionsBuilder.build();
   }
+
+  // <editor-fold desc="Public Utility">
+  /** Utility to convert {@link Configuration} to {@link BigtableDataSettings}. */
+  // TODO: Need to find a way to set BIGTABLE_USE_CACHED_DATA_CHANNEL_POOL.
+  public static BigtableDataSettings createBigtableDataSettings(Configuration configuration)
+      throws IOException {
+    Preconditions.checkState(
+        configuration.getBoolean(ENABLE_GRPC_RETRIES_KEY, RetryOptions.DEFAULT_ENABLE_GRPC_RETRIES),
+        "Disabling retries is not currently supported.");
+
+    BigtableDataSettings.Builder dataBuilder = BigtableDataSettings.newBuilder();
+
+    dataBuilder
+        .setProjectId(getValue(configuration, PROJECT_ID_KEY, "Project ID"))
+        .setInstanceId(getValue(configuration, INSTANCE_ID_KEY, "Instance ID"));
+
+    String appProfileId = configuration.get(APP_PROFILE_ID_KEY);
+    if (!Strings.isNullOrEmpty(appProfileId)) {
+      dataBuilder.setAppProfileId(appProfileId);
+    }
+
+    EnhancedBigtableStubSettings.Builder stubSettings = dataBuilder.stubSettings();
+
+    String dataHostOverride =
+        configuration.get(BIGTABLE_HOST_KEY, BigtableOptions.BIGTABLE_DATA_HOST_DEFAULT);
+    int portNumber = configuration.getInt(BIGTABLE_PORT_KEY, BigtableOptions.BIGTABLE_PORT_DEFAULT);
+    LOG.debug("API Data endpoint host %s.", dataHostOverride);
+
+    String endpoint = dataHostOverride + ":" + portNumber;
+
```

Comment:
I couldn't get what you meant by "entire endpoint"

Do you mean to get complete `hostname:port` from a single `configuration.get()`? If yes then it is not possible because Hbase doesn't bind host and port in a single config key.

Reply:
Aww, just understood what you mean.. you mean to log complete endpoint. Sorry for the trouble

Reply:
Yes, I mean:
LOG.debug("API Data endpoint host %s.", dataHostOverride);
=>
LOG.debug("endpoint host %s.", endpoint );

But it's not necessary

Reply:
Yes, that sounds good to me too. I have updated the log.